test/state.cpp

```diff
@@ -82,12 +82,38 @@ TEST_CASE_FIXTURE(StateTest, "Two Person")
 
   // Handle the Add proposal and create a Commit
   auto add = first0.add_proposal(key_packages[1]);
-  auto [commit, welcome, first1] = first0.commit(fresh_secret(), { add });
+  auto opts = State::CommitOpts{ { add }, true };
+  auto [commit, welcome, first1] = first0.commit(fresh_secret(), opts);
   silence_unused(commit);
 
   // Initialize the second participant from the Welcome
-  auto second0 =
-    State{ init_privs[1], identity_privs[1], key_packages[1], welcome };
+  auto second0 = State{
+    init_privs[1], identity_privs[1], key_packages[1], welcome, std::nullopt
+  };
+  REQUIRE(first1 == second0);
+
+  auto group = std::vector<State>{ first1, second0 };
+  verify_group_functionality(group);
+}
+
+TEST_CASE_FIXTURE(StateTest, "Two Person with external tree for welcome")
+{
+  // Initialize the creator's state
+  auto first0 = State{ group_id,          suite,           init_privs[0],
+                       identity_privs[0], key_packages[0], {} };
+
+  // Handle the Add proposal and create a Commit
+  auto add = first0.add_proposal(key_packages[1]);
+  // Don't generate RatchetTree extension
+  auto opts = State::CommitOpts{ { add }, false };
+  auto [commit, welcome, first1] = first0.commit(fresh_secret(), opts);
+  silence_unused(commit);
+
+  // Initialize the second participant from the Welcome, pass in the
+  // tree externally
+  auto second0 = State{
```

Comment:
Suggest adding another check before this of the form to verify that the State ctor throws when there's no tree:
```
CHECK_THROWS(State{..., std::nullopt});
```